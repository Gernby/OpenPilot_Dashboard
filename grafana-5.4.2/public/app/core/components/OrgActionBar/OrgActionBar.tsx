import React, { PureComponent } from 'react';
import LayoutSelector, { LayoutMode } from '../LayoutSelector/LayoutSelector';

export interface Props {
  searchQuery: string;
  layoutMode?: LayoutMode;
  onSetLayoutMode?: (mode: LayoutMode) => {};
  setSearchQuery: (value: string) => {};
  linkButton: { href: string; title: string };
  target?: string;
}

export default class OrgActionBar extends PureComponent<Props> {
  render() {
    const { searchQuery, layoutMode, onSetLayoutMode, linkButton, setSearchQuery, target } = this.props;
    const linkProps = { href: linkButton.href, target: undefined };

    if (target) {
      linkProps.target = target;
    }

    return (
      <div className="page-action-bar">
        <div className="gf-form gf-form--grow">
          <label className="gf-form--has-input-icon">
            <input
              type="text"
              className="gf-form-input width-20"
              value={searchQuery}
              onChange={event => setSearchQuery(event.target.value)}
              placeholder="Filter by name or type"
            />
            <i className="gf-form-input-icon fa fa-search" />
          </label>
          <LayoutSelector mode={layoutMode} onLayoutModeChanged={(mode: LayoutMode) => onSetLayoutMode(mode)} />
        </div>
        <div className="page-action-bar__spacer" />
        <a className="btn btn-success" {...linkProps}>
          {linkButton.title}
        </a>
      </div>
    );
  }
}
