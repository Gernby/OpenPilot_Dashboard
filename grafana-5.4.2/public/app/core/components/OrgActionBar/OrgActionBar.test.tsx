import React from 'react';
import { shallow } from 'enzyme';
import OrgActionBar, { Props } from './OrgActionBar';

const setup = (propOverrides?: object) => {
  const props: Props = {
    searchQuery: '',
    setSearchQuery: jest.fn(),
    target: '_blank',
    linkButton: { href: 'some/url', title: 'test' },
  };

  Object.assign(props, propOverrides);

  return shallow(<OrgActionBar {...props} />);
};

describe('Render', () => {
  it('should render component', () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
