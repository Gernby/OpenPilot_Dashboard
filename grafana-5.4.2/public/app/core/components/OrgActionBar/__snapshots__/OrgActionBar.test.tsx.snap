// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<div
  className="page-action-bar"
>
  <div
    className="gf-form gf-form--grow"
  >
    <label
      className="gf-form--has-input-icon"
    >
      <input
        className="gf-form-input width-20"
        onChange={[Function]}
        placeholder="Filter by name or type"
        type="text"
        value=""
      />
      <i
        className="gf-form-input-icon fa fa-search"
      />
    </label>
    <LayoutSelector
      onLayoutModeChanged={[Function]}
    />
  </div>
  <div
    className="page-action-bar__spacer"
  />
  <a
    className="btn btn-success"
    href="some/url"
    target="_blank"
  >
    test
  </a>
</div>
`;
